File: api/src/main/kotlin/com/gmtkgamejam/models/jams/Jam.kt
package com.gmtkgamejam.models.jams

import kotlinx.serialization.Serializable

@Serializable
data class Jam(
    val jamId: String,
    val name: String,
    var status: JamStatus,
    var start: String,
    var end: String,
    var styles: Map<String, String>,
    val adminIds: List<String>,
    var guildId: String,
    var channelId: String,
    var guildInviteLink: String?,
    var discordEnabled: Boolean = true
) {
    lateinit var adminInfo: Map<String, AdminInfo>
    lateinit var bgImageUrl: String
    lateinit var logoLargeUrl: String
    lateinit var logoStackedUrl: String
    lateinit var faviconUrl: String
}

File: ui/src/common/models/jam.ts
export type Jam = {
  jamId: string,
  name: string,
  status: string,
  start: string,
  end: string,
  duration: string,
  styles: {
    [key: string]: any
  },
  bgImageUrl: string,
  logoLargeUrl: string,
  logoStackedUrl: string,
  faviconUrl: string,
  adminInfo?: {[key: string]: {discordId: string, username: string}},
  guildInviteLink?: string,
}
